src/tests/serde.rs

```diff
@@ -594,3 +600,188 @@ fn test_serialize_deserialize_unsigned_numbers() {
     let doc_result: Result<Document, serde_json::Error> = serde_json::from_str(&json);
     assert!(doc_result.is_err());
 }
+
+#[test]
+fn test_unsigned_helpers() {
+    let _guard = LOCK.run_concurrently();
+
+    #[derive(Serialize)]
+    struct A {
+        #[serde(serialize_with = "serde_helpers::serialize_u32_as_i32")]
+        num_1: u32,
+        #[serde(serialize_with = "serde_helpers::serialize_u64_as_i32")]
+        num_2: u64,
+    }
+
+    let a = A { num_1: 1, num_2: 2 };
+    let doc = to_document(&a).unwrap();
+    assert!(doc.get_i32("num_1").unwrap() == 1);
+    assert!(doc.get_i32("num_2").unwrap() == 2);
+
+    let a = A {
+        num_1: u32::MAX,
+        num_2: u64::MAX,
+    };
+    let doc_result = to_document(&a);
+    assert!(doc_result.is_err());
+
+    #[derive(Serialize)]
+    struct B {
+        #[serde(serialize_with = "serde_helpers::serialize_u32_as_i64")]
+        num_1: u32,
+        #[serde(serialize_with = "serde_helpers::serialize_u64_as_i64")]
+        num_2: u64,
+    }
+
+    let b = B {
+        num_1: u32::MAX,
+        num_2: i64::MAX as u64,
+    };
+    let doc = to_document(&b).unwrap();
+    assert!(doc.get_i64("num_1").unwrap() == u32::MAX as i64);
+    assert!(doc.get_i64("num_2").unwrap() == i64::MAX);
+
+    let b = B {
+        num_1: 1,
+        num_2: i64::MAX as u64 + 1,
+    };
+    let doc_result = to_document(&b);
+    assert!(doc_result.is_err());
+}
+
+#[test]
+fn test_datetime_helpers() {
+    let _guard = LOCK.run_concurrently();
+
+    #[derive(Deserialize, Serialize)]
+    struct A {
+        #[serde(deserialize_with = "serde_helpers::deserialize_bson_datetime_from_ext_json")]
+        pub date: DateTime,
+    }
+
+    let date = DateTime(chrono::DateTime::from_str("2020-06-09 10:58:07.095 UTC").unwrap());
+    let a = A { date };
+    let bson = to_bson(&a).unwrap();
+    let value: Value = bson.into();
+    let a: A = serde_json::from_value(value).unwrap();
+    assert_eq!(a.date, date);
+
+    #[derive(Deserialize)]
+    struct B {
+        #[serde(deserialize_with = "serde_helpers::deserialize_chrono_datetime_from_ext_json")]
+        pub date: chrono::DateTime<chrono::Utc>,
+    }
+
+    let date = r#"
+    {
+        "date": {
+                "$date": {
+                    "$numberLong": "1591700287095"
+                }
+        }
+    }"#;
+    let json: Value = serde_json::from_str(&date).unwrap();
+    let b: B = serde_json::from_value(json).unwrap();
+    let expected: chrono::DateTime<chrono::Utc> =
+        chrono::DateTime::from_str("2020-06-09 10:58:07.095 UTC").unwrap();
+    assert_eq!(b.date, expected);
+
+    #[derive(Deserialize, Serialize)]
+    struct C {
+        #[serde(
+            serialize_with = "serde_helpers::serialize_iso_string_as_datetime",
```

Comment:
Serde provides a way to specify both serialization and deserialization with the same attribute [through modules](https://serde.rs/field-attrs.html#with). I think this is a good case for it; in addition to defining the functions that already exist, we could make a module that just defines thin wrappers around each of those two functions so that users who want full roundtripping between ISO strings and DateTime can do that with one attribute.